test/server/server_test.cc

```diff
@@ -465,6 +466,39 @@ TEST_P(ServerInstanceImplTest, ZipkinHttpTracingEnabled) {
   EXPECT_NE(nullptr, dynamic_cast<Tracing::HttpTracerImpl*>(tracer()));
 }
 
+class TestObject : public ProcessObject {
+public:
+  TestObject() {}
+  void setFlag(bool value) { boolean_flag_ = value; }
+
+  bool boolean_flag_ = true;
+};
+
+TEST_P(ServerInstanceImplTest, WithProcessContext) {
+  TestObject object;
+
+  options_.config_path_ = TestEnvironment::temporaryFileSubstitute(
+      "test/server/empty_bootstrap.yaml", {{"upstream_0", 0}, {"upstream_1", 0}}, version_);
+  thread_local_ = std::make_unique<ThreadLocal::InstanceImpl>();
+
+  auto server = std::make_unique<InstanceImpl>(
```

Comment:
I think it might make sense to reuse the parent `initialize` for this, to reduce boilerplate. Maybe make it possible to set the `ProcessContext` on the parent, which can then feed into `initialize()`?